state/protocol/snapshots/dynamic_bootstrap.go

```diff
@@ -0,0 +1,118 @@
+package snapshots
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol"
+)
+
+var ErrSnapshotPhaseMismatch = errors.New("snapshot does not contain a valid sealing segment")
+var ErrSnapshotHistoryLimit = fmt.Errorf("reached the snapshot history limit")
+
+// GetDynamicBootstrapSnapshot will return a valid snapshot for dynamic bootstrapping
```

Comment:
```suggestion
// GetDynamicBootstrapSnapshot returns `refSnapshot` if it is valid for use in dynamic bootstrapping.
// Otherwise returns an error. (Effectively this validates that the input snapshot can be used in dynamic bootstrapping.)
```